package.json

```diff
@@ -185,7 +185,7 @@
     "webpack-http-server": "^0.5.0"
   },
   "peerDependencies": {
-    "typescript": ">= 4.7.x <= 5.2.x"
+    "typescript": ">= 4.7.x <= 5.3.x"
```

Comment:
is there a reason why there is strict enforcement of a version? if there are incompatibilities between versions, TypeScript will be more than happy tell about it on its own by failing the build. There is questionable value and a lots of inconvenience in explicit TypeScript version management, most libraries don't do that.

Reply:
https://github.com/mswjs/msw/pull/1874/files#r1402712831 answered it

Reply:
In my experience, there are incompatible behaviors across TypeScript versions. I recall distinctly that the same code base compiled on 4.6, failed on 4.7, and compiled again on 4.8. That's the main reason the version range is strict and was locked. 

With the version testing we have in place now, it's much more reliable to open up this version range as we will catch compilation errors before we merge the changes, which is always what we want. 

As Matt has said, we will merge this to unblock people (once all the pending bug fixes are released first) and then address the version range unlock as the next step.

Reply:
@kettanaito TS 5.4 is out, and msw again is the only dependency blocking upgrade for us.
Maybe now is a good time to address this issue conclusively?..

Reply:
@kibertoad, I'm opening https://github.com/mswjs/msw/pull/2077 to solve this for good. Thanks for helping this happen.